server/src/main/java/io/unitycatalog/server/persist/RepositoryFactory.java

```diff
@@ -0,0 +1,36 @@
+package io.unitycatalog.server.persist;
+
+import io.unitycatalog.server.persist.utils.FileOperations;
+import io.unitycatalog.server.utils.ServerProperties;
+import java.util.HashMap;
+import java.util.Map;
+import lombok.Getter;
+import org.hibernate.SessionFactory;
+
+public class RepositoryFactory {
+  private final Map<Class<?>, Object> repositories = new HashMap<>();
+  @Getter private final SessionFactory sessionFactory;
+  @Getter private final FileOperations fileOperations;
+
+  public RepositoryFactory(SessionFactory sessionFactory, ServerProperties serverProperties) {
+    this.sessionFactory = sessionFactory;
+    this.fileOperations = new FileOperations(serverProperties);
+    initializeRepositories();
+  }
+
+  public void initializeRepositories() {
+    repositories.put(CatalogRepository.class, new CatalogRepository(this, sessionFactory));
+    repositories.put(SchemaRepository.class, new SchemaRepository(this, sessionFactory));
+    repositories.put(TableRepository.class, new TableRepository(this, sessionFactory));
+    repositories.put(VolumeRepository.class, new VolumeRepository(this, sessionFactory));
+    repositories.put(UserRepository.class, new UserRepository(this, sessionFactory));
+    repositories.put(MetastoreRepository.class, new MetastoreRepository(this, sessionFactory));
+    repositories.put(FunctionRepository.class, new FunctionRepository(this, sessionFactory));
+    repositories.put(ModelRepository.class, new ModelRepository(this, sessionFactory));
+  }
+
+  @SuppressWarnings("unchecked")
+  public <T> T getRepository(Class<T> repositoryClass) {
```

Comment:
also .. i dont get this model. There is a fixed number of repositories that we are tracking within this class. why put them in a map and then make the caller specify the class explicitly and all this complex calling? why not just make simply have fields+getters for each repo? then calling reduces from 
`repositoryFactory.getRepository(UserRepository.class)` to simply `repositoryFactory.getUserRepository()`